src/matchers/element/toMatchSnapshot.ts

```diff
@@ -0,0 +1,43 @@
+import { SnapshotClient } from '@vitest/snapshot'
+import { WdioElementMaybePromise } from '../../types.js'
+import { executeCommandBe } from '../../utils.js'
+
+
+let _client: SnapshotClient
+
+function getSnapshotClient(): SnapshotClient {
+    if (!_client) {
+        _client = new SnapshotClient({
+            isEqual: (received, expected) => {
+                return received == expected
+            },
+        })
+    }
+    return _client
+}
+
+export function toMatchSnapshot(received: WdioElementMaybePromise, options: ExpectWebdriverIO.CommandOptions = {}) {
+    this.expectation = this.expectation || 'matchSnapshot'
+    return executeCommandBe.call(this, received, fnToMatchSnapShot, options)
+}
+
+async function fnToMatchSnapShot(element: WebdriverIO.Element, options?: ExpectWebdriverIO.CommandOptions): Promise<boolean> {
+    console.log(options);
+    try {
+        getSnapshotClient().assert({
+            received: element,
```

Comment:
This would snapshot the `WebdriverIO.Element` rather than its HTML code right?

Reply:
Actually it can be any object. Here I'm using as WebdriverIO.Element

Reply:
I don't think snapshotting the `WebdriverIO.Element` is much useful, we should snapshot its HTML content.

Reply:
I'm about to use it this way:

await expect({a:"a"}).toMatchSnapshot()
await expect($('#flash')).toMatchSnapshot()

so depending on what is the received Type we can handle it diff way

Reply:
How about we keep `await expect({ a: 123 }).toMatchSnapshot()` to snapshot the object as in other frameworks and create a `await expect($('#flash')).toMatchHTMLSnapshot()` where we call `getHTML` from the object and snapshot that. Wdyt?

I prefer to keep things more implicit than hiding some *magic* beheind it.